### ISSUE TITLE
[Package Request]: Adding Tencent iOA 

### ISSUE THREAD

**Issue by zzhan353** [microsoft/winget-pkgs#159654]
### How can we help?

I would like someone else to build the manifest.

### Please read and ensure the following

- [X] The installer meets the above requirements

### Please provide the following information

Download Page Url:https://ioa.tencent.com/?download_url=https%3A%2F%2Fdownload.ioa.tencent.com%2Fioav5_setup5.4.20.142.exe?download_name=iOA
Publisher: Tencent
Package Name:Tencent iOA
Description:Tencent Endpoint Security Management Platform
Package Version:5.4.20.142
Installer URL:https://ioa.tencent.com/api/download?platform=windows

**Comment by Dragon1573** [comment]
Sorry @zzhan353, this issue might should be closed as "not planned".

The [installer URL](https://ioa.tencent.com/api/download?platform=windows) you provided will redirects to a [login page](https://login.woa.tencent.com/modules/passport/signin.ashx) with parameters `oauth`, `appkey` and `url`. The [download page URL](https://ioa.tencent.com/?download_url=https%3A%2F%2Fdownload.ioa.tencent.com%2Fioav5_setup5.4.20.142.exe?download_name=iOA) you provided will redirects to a [direct link](https://download.ioa.tencent.com/ioav5_setup5.4.20.142.exe), with an `HTTP 403 Forbidden` response from Nginx server.

Seems the installer file requires authentication to download. `winget.exe` does not support such kind of packages. It would be better hosting the package on your own by using _winget.pro_ or _WinGetty_.

**Comment by Dragon1573** [comment]
I've searched keywork "腾讯iOA" on Google, and get following links. Direct (without authentication required) links are provided on them, but I can't sure if they are the same from [official](https://ioa.tencent.com/), could you please download and hash them to verify?

- https://ioa.intlgame.com/d
  - It's from a third-party domain, seems not related to Tencent.
- https://ioa.cloud.tencent.com/download
  - It's the SaaS version of _Tencent iOA_, seems similar to `Tencent.WXWork` (企业微信公有云版本) and `Tencent.WXWorkLocal` (企业微信私有部署).

**Comment by zzhan353** [comment]
Hi May I know why not use the one from the official website? It is the first one if you search tencent ioa in google. 
Thanks.

**Comment by zzhan353** [comment]
click 'Windows' on the page of https://ioa.tencent.com/, the that installer would be the expected one.

**Comment by Dragon1573** [comment]
> Click <kbd>Windows</kbd> on the page of https://ioa.tencent.com/, the installer would be the expected one.

![recording](https://github.com/microsoft/winget-pkgs/assets/49941141/fbd3f1e4-9529-4aa0-8896-889b640052d7)

Authentication is required to download the package. `winget.exe` does not support this.

**Comment by Dragon1573** [comment]
```text
# Download page URL: https://ioa.intlgame.com/d
# Version number: 106.6.11242.5515
# URL associated with the download button: https://ioa.intlgame.com/downloadClient?ver=win
# Direct link: https://ioa.intlgame.com/store/PCMgr_Setup_cfg(ioa.intlgame.com@27800).exe

14:54:04 D:\...\winget-pkgs  [master ≡] 2ms pwsh> winget hash 'D:\Dragon1573\Downloads\PCMgr_Setup_cfg(ioa.intlgame.com@27800).exe'
InstallerSha256: eb5a7da666d8d2176c4f05391a5c22c6be402c7d0ff3c30c2918d53d7c52ab82
```

```text
# Download page URL: https://ioa.cloud.tencent.com/download
# Version number: 209.3.20968.62002
# No direct link to this package!

15:00:28 D:\...\winget-pkgs  [master ≡] 26ms pwsh> winget hash D:\Dragon1573\Downloads\PCMgr_Setup.exe
InstallerSha256: 6034727ad60303ba7aa893c50cfc5e013a8d3e33a271329936feeb7f3beb8541
```

**Comment by zzhan353** [comment]
I see. I guess I am in Tencent's networking so it does not require me to login and directly download that package. Is it possible I provide the installer file? W're also trying to set up a local repo but the complexity is higher than expected because of the networking constraints. Thanks

**Comment by Dragon1573** [comment]
> Is it possible I provide the installer file?

NO, you don't. I haven't use _Tencent iOA_ (both local ver or SaaS ver), so I cant't say much about it. In my opinion, leaking the installer file to external area may do harm to your company.

> W're also trying to set up a local repo but the complexity is higher than expected because of the networking constraints.

Okay ... Sounds complicated. I'm in an open network environment so I can use official `msstore` and `winget` source. I haven't use local `winget` source before.

What about using _WinGetty_? It provides ability to deploy locally. It may help in your situation.

**Comment by zzhan353** [comment]
Hmm, I checked with iOA admin and confirmed that they do not provide the download url without login. Yup, I am using WinGetty in the local demo but it will be complex to put it into tencent's production networking environment. Thanks anyway.
